OneSignalSDK/onesignal/src/main/java/com/onesignal/NotificationPayloadProcessorHMS.java

```diff
@@ -0,0 +1,74 @@
+package com.onesignal;
+
+import android.app.Activity;
+import android.content.Intent;
+import android.os.Bundle;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import static com.onesignal.GenerateNotification.BUNDLE_KEY_ACTION_ID;
+
+class NotificationPayloadProcessorHMS {
+
+    static void handleHmsNotificationOpenIntent(@NonNull Activity activity, @Nullable Intent intent) {
+        OneSignal.setAppContext(activity);
+        if (intent == null)
+            return;
+
+        JSONObject jsonData = covertHmsOpenIntentToJson(intent);
+        if (jsonData == null)
+            return;
+
+        handleProcessJsonOpenData(activity, jsonData);
+    }
+
+    // Takes in a Notification Open Intent fired from HMS Core and coverts it to an OS formatted JSONObject
+    // Returns null if it is NOT a notification sent from OneSignal's backend
+    private static @Nullable JSONObject covertHmsOpenIntentToJson(@Nullable Intent intent) {

```

Comment:
So that its consistent should we capitalize `HMS` in `covertHmsOpenIntentToJson`?